scrape/scrape.go

```diff
@@ -1310,8 +1312,20 @@ loop:
 		if ok {
 			err = app.AddFast(ce.ref, t, v)
 			_, err = sl.checkAddError(ce, met, tp, err, &sampleLimitErr, &appErrs)
+			switch err {
+			case nil:
+				if hasExemplar := p.Exemplar(&e); hasExemplar {
+					if !e.HasTs {
+						e.Ts = t
+					}
+					if err := app.AddExemplarFast(ce.ref, e); err != nil {
+						if err != storage.ErrDuplicateExemplar {
+							level.Debug(sl.l).Log("msg", "Unexpected error", "error", err, "seriesLabels", ce.lset, "exemplar", e)
+						}
+					}
+				}
```

Comment:
nit: the level of nesting here seems super undesirable and hard to reason about.  I suggest breaking it out into a separate function and straight-lining the error handling.

not nit: any error from app.AddExemplarFast(ce.ref, e) is being swallowed here.  That doesn't seem intentional?

```suggestion
append := func(ref, t, v) {
	err = app.AddFast(ce.ref, t, v)
	_, err = sl.checkAddError(ce, met, tp, err, &sampleLimitErr, &appErrs)
	if err == storage.ErrNotFound {
		ok = false
		return
	}
	
	if hasExemplar := p.Exemplar(&e); !hasExemplar {
		return
	}
	
	if !e.HasTs {
		e.Ts = t
	}
	
	err := app.AddExemplarFast(ce.ref, e)
	if err != nil && if err != storage.ErrDuplicateExemplar {
		level.Debug(sl.l).Log("msg", "Unexpected error", "error", err, "seriesLabels", ce.lset, "exemplar", e)
	}
}

...

append(ce.ref, t, v)
```